go/genkit/genkit.go

```diff
@@ -35,65 +35,100 @@ import (
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
 )
 
+// Plugin is a common interface for plugins.
+type Plugin interface {
```

Comment:
wondering if this is better as an abstract class.We plan to add `ResolveAction(ctx context.Context, g *Genkit, n: string)` method, which should be optional.

Reply:
As discussed, dynamic resolution will be implemented as a separate `DynamicPlugin` interface that we will try to cast to during generation.